resources/assets/js/ready.js

```diff
@@ -0,0 +1,7 @@
+export const ready = (fn) => {
+  if (document.readyState !== 'loading') {
+    fn();
+  } else {
+    document.addEventListener('DOMContentLoaded', fn);
+  }
+}
```

Comment:
i would use the code from jquery
https://github.com/jquery/jquery/blob/main/src/core/ready.js#L57-L78
```js
function ready(callback) {
  // The ready event handler and self cleanup method
  function completed() {
    document.removeEventListener('DOMContentLoaded', completed);
    window.removeEventListener('load', completed);
    callback();
  }

  // Catch cases where onDocumentLoaded() is called after the browser event has already occurred.
  if (document.readyState !== 'loading') {
    // Handle it asynchronously to allow scripts the opportunity to delay ready
    window.setTimeout(callback);
  } else {
    // Use the handy event callback
    document.addEventListener('DOMContentLoaded', completed);

    // A fallback to window.onload, that will always work
    window.addEventListener('load', completed);
  }
}
```

Reply:
arrow-function? :trollface:

Reply:
`DOMContentLoaded` is supported by all current browsers. Fallback options are included in jQuery to support IE <= 8. I would mandate against copying over jQuery legacy stuff.

Reply:
ok but i would rename `fn` to `callback`.
And should we do this with the `window.setTimeout(callback);`?

Reply:
> ok but i would rename fn to callback.

:man_shrugging: can do

> And should we do this with the window.setTimeout(callback);?

Any specific reason for this?

Reply:
i only thought if jquery did it they would have a reason for it.
i think this is that in both cases the callback gets pushed at the end of the event loop (and not get executed directly) and behaves the same way

Reply:
jQuery is from the very past :slightly_smiling_face: The snippet above is already battle proven.